bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/SerialBridgeThingHandler.java

```diff
@@ -31,31 +37,69 @@
 @NonNullByDefault
 public class SerialBridgeThingHandler extends KNXBridgeBaseThingHandler {
 
-    private final SerialClient client;
+    private @Nullable SerialClient client = null;
+    private @Nullable Future<?> initJob = null;
+
+    private final Logger logger = LoggerFactory.getLogger(SerialBridgeThingHandler.class);
 
     public SerialBridgeThingHandler(Bridge bridge) {
         super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        // create new instance using current configuration settings;
+        // when a parameter change is done from UI, dispose() and initialize() are called
         SerialBridgeConfiguration config = getConfigAs(SerialBridgeConfiguration.class);
         client = new SerialClient(config.getAutoReconnectPeriod(), thing.getUID(),
                 config.getResponseTimeout().intValue(), config.getReadingPause().intValue(),
                 config.getReadRetriesLimit().intValue(), getScheduler(), config.getSerialPort(), config.useCemi(),
                 this);
-    }
 
-    @Override
-    public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
-        client.initialize();
+        // delay actual initialization, allow for longer runtime of actual initialization
+        initJob = scheduler.submit(() -> {
+            initializeLater();
+        });
+    }
+
+    public void initializeLater() {
+        final var tmpClient = client;
+        if (tmpClient != null) {
+            tmpClient.initialize();
+        }
     }
 
     @Override
     public void dispose() {
-        client.dispose();
+        final var tmpInitJob = initJob;
+        if (tmpInitJob != null) {
+            while (!tmpInitJob.isDone()) {
+                logger.trace("Bridge {}, shutdown during init, trying to cancel", thing.getUID());
+                tmpInitJob.cancel(true);
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    logger.trace("Bridge {}, cancellation interrupted", thing.getUID());
+                }
```

Comment:
Generally, we cancel the job without waiting.
Do you really need to wait for the cancel to complete ?
By the waay, I am not fully sure that looping on isDone() is correct. To be verified.

Reply:
Looping is probably unnecessary, as the platform docs state the following for ```cancel()```:

> "After this method returns, subsequent calls to isDone() will always return true."

But same here, I would play it safe for the release. The code is in use for IP bridge and has already some test mileage.

Reply:
This is new code and in SerialBridgeThingHandler class, so I guess it is not already used for the IP handler.
So why adding something not clean? I do not understand your argument, sorry.

Reply:
At least do it as expected here, even if it is not done properly elsewhere.

Reply:
Ok, lets do the following: I change the while to an if. Then it should be more clear and actually do the same in case the doc is correct. It will cancel the init job if it is not yet started, and try to cancel it in case it is already running, and do nothing if it already completed. WDYT?

Just a sidenote: What I actually meant with test mileage is the hours of debugging I spent to get the SECURETUNNEL for IP working properly - without eating up all tunnel connections due to improperly closing. The code for initialization of the thing should be quite similar for IP and serial, but for the serial gateway I did not dare to change anything without having a device at hand for testing. Delayed initialization was already required due to IP secure for IP, but not yet in serial gateways. Data secure - both relevant to serial and IP gateways - is still under development (open PR, needs some more work). So actually the code I picked for the PR here was already available.

Reply:
Why keeping a sleep of 1s just after cancelling the thread?

Reply:
To let both sides properly disconnect. Not sure if this is a hard requirement for serial, but it will not hurt. I have only one device at hand for testing.
Just think about when this code will be running: only if the thing is quickly reloaded, typically when someone saves the text config more than once.```initialize()``` schedules the initialization (which typically starts very soon after this), and the initialization is finished within a few sec.

Reply:
@kaikreuzer : as naintainer of this binding, are you ok with this unusual code pattern in disposr() method.  If you are, you can merge the PR.

Reply:
While not looking beautifully, I think with the if instead of the while, we can accept it. 😉